Orange/data/io.py

```diff
@@ -337,248 +349,169 @@ def open(filename, *args, **kwargs):
+                if name == new_name:
+                    var = col_type.make(new_name, **col_type_kwargs)
                 else:
-                    # Never use existing for un-named variables
-                    var = coltype(next(NAMEGEN), **coltype_kwargs)
+                    var = col_type(new_name, **col_type_kwargs)
+                # also store any attributes in the third row (beside the role declaration)
                 var.attributes.update(flag.attributes)
-                domain_vars.append(var)
-
-                # Reorder discrete values to match existing variable
-                if var.is_discrete and not var.ordered:
-                    new_order, old_order = var.values, coltype_kwargs.get('values', var.values)
-                    if new_order != old_order:
-                        offset = len(new_order)
-                        column = values if data.ndim > 1 else data
-                        column += offset
-                        for i, val in enumerate(var.values):
-                            try: oldval = old_order.index(val)
-                            except ValueError: continue
-                            bn.replace(column, offset + oldval, new_order.index(val))
-
-            if coltype is TimeVariable:
-                # Re-parse the values because only now after coltype.make call
-                # above, variable var is the correct one
-                values = [var.parse(i) for i in orig_values]
-
-            # Write back the changed data. This is needeed to pass the
-            # correct, converted values into Table.from_numpy below
-            try: data[:, col] = values
-            except IndexError: pass
-
-        domain = Domain(attrs, clses, metas)
-
-        if not data.size:
-            return Table.from_domain(domain, 0)
-
-        table = Table.from_numpy(domain,
-                                 data[:, Xcols].astype(float, order='C'),
-                                 data[:, Ycols].astype(float, order='C'),
-                                 data[:, Mcols].astype(object, order='C'),
-                                 data[:, Wcols].astype(float, order='C'))
-        return table
+
+                role_vars[col_role].append(var)
+                # strip whitespace from string/string-like columns (np.object_ in pandas)
+                if np.issubdtype(contents[col_idx], np.object_):
+                    result[var.name] = contents[col_idx].str.strip()
+                else:
+                    result[var.name] = contents[col_idx]
+
+            domain = Domain(role_vars['x'], role_vars['y'], role_vars['meta'])
+            result = Table.from_dataframe(result, domain, reindex=True, weights=weight_column)
+        else:
+            # there is no header, just read the file
+            # and pass it to the proper constructor to infer columns
+            result = Table.from_dataframe(self.read_contents(skiprows=0))
```

Comment:
I was completely sure I added that, turns out I only did the one with a known domain. Thank you very much.